### PR TITLE
[SEDONA-558] Fix and improve SedonaPyDeck behavior

### PR DESCRIPTION
**Pull request by furqaankhan**
## Did you read the Contributor Guide?

- Yes, I have read the [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-558. The PR name follows the format `[SEDONA-XXX] my subject`

### Current implementation

- Adds the optional parameters `map_style`, and `map_provider` as data members to the pdk map object. 

     eg: map_p.map_style = map_style

- The current behavior of PyDeck doesn't update the map object with the additional configuration added via data members.

### Behavior of PyDeck

- The default `map_style` is 'dark' and `map_provider` is 'carto'. If you configure `map_style` to 'satellite' it will not change the `map_provider` to Mapbox or Google Maps API. 

## What changes were proposed in this PR?

- Expose `api_key` parameter of PyDeck
- Create a new method to create the PyDeck object to handle the default behavior of PyDeck.
- Default to Mapbox map provider if the `map_style` is set to 'satellite' and if `map_provider` is Carto (the default).

## How was this patch tested?

- Passed existing tests.

## Did this PR include necessary documentation updates?

- Yes, I have updated the documentation.

### CODE REVIEW

**Review by jiayuasu — CHANGES_REQUESTED** [review]
Why is this PR changing these files?

**Review by jiayuasu — CHANGES_REQUESTED** [review]
Is it possible to add some test cases?

File: python/tests/sql/test_dataframe_api.py
Comment (jiayuasu) [inline python/tests/sql/test_dataframe_api.py:1]: Why is this PR changing these files?

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: python/tests/sql/test_dataframe_api.py
Comment (furqaankhan) [reply python/tests/sql/test_dataframe_api.py:1]: They are phony changes, these changes have been merged before.

**Comment by furqaankhan** [comment]
To test if the API key has been added successfully [is not possible](https://github.com/visgl/deck.gl/blob/b738b66e97fef8995daa8b5d9470dfbfe2fdc9c2/bindings/pydeck/pydeck/bindings/json_tools.py#L8-L17). However, I can add tests for other behavior changes.

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**